src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -326,31 +327,59 @@ private AnalyzerSettings GetLanguageSpecificSettings(AnalysisConfig config)
 
         /// <summary>
         /// Merges and returns the supplied list of file paths. In case of duplicate
-        /// // file *names* (not full paths), the path from the primary list is used.
+        /// file *names* (not full paths) of SonarAnalyzers, the path from the sonarConfiguration list is used.
         /// </summary>
-        private string[] MergeFileLists(IEnumerable<string> primaryList, IEnumerable<string> secondaryList)
+        private string[] RemoveDuplicateSonarAnalyzerReferences(IEnumerable<string> sonarConfiguration, IEnumerable<string> userProvidedConfiguration)
         {
-            var nonNullPrimary = primaryList ?? Enumerable.Empty<string>();
-            var nonNullSecondary = secondaryList ?? Enumerable.Empty<string>();
+            var nonNullPrimary = sonarConfiguration ?? Enumerable.Empty<string>();
+            var nonNullSecondary = userProvidedConfiguration ?? Enumerable.Empty<string>();
 
-            var duplicates = GetEntriesWithMatchingFileNames(nonNullPrimary, nonNullSecondary);
+            var sonarAnalyzerDuplicates = nonNullSecondary
+               .Where(x => Regex.IsMatch(GetFileName(x), @"sonaranalyzer\..*\.dll", RegexOptions.IgnoreCase))
```

Comment:
and open question: should we really be that restrictive to accepted third party analyzer names? 

This means that we deny customers to create analyzers that contain SonarAnalyzer in their name... for example `MyInternalSonarAnalyzer.dll` will be removed (and if they write their analyzers expecially for SonarQube integration, it's quite counterintuitive` (maybe ask @tom-howlett-sonarsource for a 2nd opinion)

and if they have their analyzers put in a folder named `sonaranalyzers` (`/a/b/sonaranalyzer/foo.dll`), again they won't be able to do that and will have weird behavior because of that

Isn't it safer to have a hardcoded list of our analyzer libraries (we don't add many)?

Reply:
Yes, I think we should at least document it (not sure where we talk about 3rd party analyzers). If there is a fixed list of ours that we could it would be nice not to complicate the documentation!

Reply:
I added the hardcoded list of analyzers that we currently ship.
I would also like to clarify that with the regex solution we would not exclude usage of  analyzers in sonaranalyzer folders as it was first getting the FileName and the regex was applied on that.

Reply:
(sorry, my bad I made an untested assumption)

Reply:
No worries! It makes sense to add also a test case for that. Thanks for the suggestion.